src/Atmos/Model/linear.jl

```diff
@@ -229,7 +233,11 @@ function source!(
 )
     if direction isa VerticalDirection || direction isa EveryDirection
         ∇Φ = ∇gravitational_potential(lm.atmos.orientation, aux)
-        source.ρu -= state.ρ * ∇Φ
+        if lm.atmos.ref_state isa HydrostaticState
+            source.ρu -= (state.ρ - aux.ref_state.ρ) * ∇Φ
```

Comment:
Won't this make the operator non-linear ? Our ARK schemes could be changed to handle systems of the form `L * x + b = c` by moving `b` to the right hand side. I have an old branch that has this functionality but was never tested (mw/affine-ark).

Reply:
Yes, this would need some checking.

Reply:
It'll remain linear (`aux.ref_state.ρ` is constant). But it does add an intercept term.

Reply:
Right, I meant linearity in the linear algebra sense (which is what we currently assume).

Reply:
How is discrete balance defined?

Reply:
Would this be easier if we changed our state variable from ρ to δρ = ρ - ρ_ref?

Reply:
Basically that the two added terms will cancel out to machine precision using DG discretization.

Reply:
Discrete balance means that \rho = -1/g dp/dz discretely. This is being implemented from what I know. 

Working with perturbation variables (for pressure and density) would help here. But not for the acoustic waves (the intercept there would arise anyway). And you'd need to go back to native variables for all thermodynamics, microphysics etc. I'd prefer to avoid that back and forth between variables.

Reply:
We'd also need to account for this when forming the linear system.